### PR TITLE
Staging channel events

### PR DESCRIPTION
**Pull request by AmmadRehmatCUS**
Wifi 885, Wifi 888 : Events channel events switching due to radar detection and high interference.

Signed-off-by: Ammad Rehmat ammad.rehmat@connectus.ai

### CODE REVIEW

File: feeds/wifi-trunk/hostapd/patches/901-hapd-ubus-channel-switch.patch [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
++	struct timespec ts;
++	clock_gettime(CLOCK_REALTIME, &ts);
++	rec = os_zalloc(sizeof(struct hostapd_chan_event_list));
++
++	rec->records.reason = reason;
++	rec->records.band = iface->freq;
++	rec->records.ts = ts;
++	rec->records.freq = freq;
++
++	//if (list_empty(&chan_events->list)) {
```
Comment (blogic) [inline feeds/wifi-trunk/hostapd/patches/901-hapd-ubus-channel-switch.patch:124]: dead code, how often do i need to say this ..

File: feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c [hunk 2]
```diff
@@ -622,6 +642,99 @@ static void ubus_collector_bss_cb(struct ubus_request *req, int type,
...
+{
+
+	int error = 0;
+	int rem = 0;
+	struct blob_attr *tb_chan_event_lst[__CHANNEL_SWITCH_EVENT_MAX] = {};
+	struct blob_attr *tb_chan_event_table[__CHANNEL_SWITCH_MAX] = {};
+	struct blob_attr *tb_chan_event_tbl = NULL;
+	dpp_event_record_t *channel_event_list = NULL;
+	dpp_event_record_channel_switch_t *channel_event_record = NULL;
+	//ds_dlist_iter_t record_iter;
```
Comment (blogic) [inline feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c:656]: more dead code

Comment (AmmadRehmatCUS) [reply feeds/wlan-ap/opensync/src/src/sm/src/ubus_collector.c]: fixed

File: feeds/wifi-trunk/hostapd/patches/901-hapd-ubus-channel-switch.patch (hunk 1 above)
Comment (AmmadRehmatCUS) [reply feeds/wifi-trunk/hostapd/patches/901-hapd-ubus-channel-switch.patch:124]: fixed

**Comment by blogic** [comment]
please fold the patches into a single one and add proper annotation

**Comment by AmmadRehmatCUS** [comment]
> please fold the patches into a single one and add proper annotation

commits have been squashed into a single one, if that's what you were pointing to.

**Comment by yashvardhan-netex** [comment]
@RickSommerville @blogic  This pull request should not be merged for now.  Needs revision due to client event refactoring.

**Comment by RickSommerville** [comment]
This is obsolete and has been replaced by pull request  #150

**Pull request closed without merging**